File: uhealthy/src/main/java/com/fmu/uhealthy/domain/Patient.java

package com.fmu.uhealthy.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import javax.persistence.*;
import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity(name="patient")
public class Patient {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name")
    private String name;

    private String documentNumber;

    @CreationTimestamp
    private LocalDate createDate;

    @Column(name = "birth_date")
    private LocalDate birthDate;

    @ManyToOne
    @JoinColumn(name = "gender_id")
    private Gender gender;

    @JsonIgnore
    @OneToOne(mappedBy = "patient", fetch = FetchType.LAZY)
    private User user;

    @Transient
    private String profileImage;

    @Transient
    private String email;

    @Transient
    private String phone;

    @SuppressWarnings("unused")
    public String getProfileImage() {
        return user!= null ? user.getProfileImage() : "";
    }

    @SuppressWarnings("unused")
    public String getEmail() {
        return user!= null ? user.getLogin() : "";
    }

    @SuppressWarnings("unused")
    public String getPhone() {
        return user!= null ? user.getPhone() : "";
    }
}

File: front/lib/model/patient.dart
import 'gender.dart';

class Patient {
  final int? id;
  final String? name;
  final String? documentNumber;
  final DateTime? createDate;
  final DateTime? birthDate;
  final Gender? gender;
  final String? profileImage;
  final String? phone;
  final String? email;

  Patient(this.id, this.name, this.documentNumber, this.createDate,
      this.birthDate, this.gender, this.profileImage, this.phone, this.email);

  static convertPatient(patient) {
    if (patient != null) {
      return Patient(
          patient['id'],
          patient['name'],
          patient['documentNumber'],
          DateTime.parse(patient['createDate']),
          DateTime.parse(patient['birthDate']),
          Gender.convertGender(patient['gender']),
          patient['profileImage'],
          patient['phone'],
          patient['email']);
    } else {
      return Patient(null, null, null, null, null, null, null, null, null);
    }
  }

  Map<String, dynamic> toJson() {
    final _data = <String, dynamic>{};
    _data['id'] = id;
    _data['name'] = name;
    _data['documentNumber'] = documentNumber;
    _data['createDate'] = createDate?.toIso8601String();
    _data['birthDate'] = birthDate?.toIso8601String();
    _data['gender'] = gender?.toJson();
    _data['profileImage'] = profileImage;
    _data['phone'] = phone;
    _data['email'] = email;
    return _data;
  }
}
